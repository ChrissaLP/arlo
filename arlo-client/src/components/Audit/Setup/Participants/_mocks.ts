const jurisdictionFile = new File(
  [
    'Jurisdiction","Admin Email"',
    '"Death Star","wtarkin@empire.gov"',
    '"Hoth","admin@rebelalliance.ninja"',
    '"Tatooine","jabba@hutt.biz"',
  ],
  'jurisdictions.csv',
  { type: 'text/csv' }
)

export default jurisdictionFile
