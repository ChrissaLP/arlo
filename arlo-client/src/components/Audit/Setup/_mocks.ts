import { ElementType } from '../../../types'
import { setupStages } from './index'
import { ISidebarMenuItem } from '../../Atoms/Sidebar'

const relativeStages = (
  stage: ElementType<typeof setupStages>,
  state: ISidebarMenuItem['state'] = 'live'
) => {
  const prevTitle = setupStages[setupStages.indexOf(stage) - 1]
  const prevStage: ISidebarMenuItem = {
    title: prevTitle,
    active: false,
    activate: jest.fn(),
    state,
  }
  const nextTitle = setupStages[setupStages.indexOf(stage) + 1]
  const nextStage: ISidebarMenuItem = {
    title: nextTitle,
    active: false,
    activate: jest.fn(),
    state,
  }
  return { prevStage, nextStage }
}

export default relativeStages
