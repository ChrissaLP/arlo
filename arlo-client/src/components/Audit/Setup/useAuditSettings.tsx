import { useEffect, useCallback, useState } from 'react'
import { api, checkAndToast } from '../../utilities'
import { IAuditSettings, IErrorResponse } from '../../../types'

const defaultValues: IAuditSettings = {
  state: null,
  electionName: null,
  online: null,
  randomSeed: null,
  riskLimit: null,
}

type TNewSettings =
  | {
      state: IAuditSettings['state']
    }
  | {
      electionName: IAuditSettings['electionName']
      online: IAuditSettings['online']
      randomSeed: IAuditSettings['randomSeed']
      riskLimit: IAuditSettings['riskLimit']
    }

const useAuditSettings = (
  electionId: string
): [IAuditSettings, (arg0: TNewSettings) => void] => {
  const [settings, setSettings] = useState(defaultValues)

  const getSettings = useCallback(async (): Promise<IAuditSettings> => {
    const settingsOrError: IAuditSettings | IErrorResponse = await api(
      `/election/${electionId}/settings`
    )
    if (checkAndToast(settingsOrError)) {
      return defaultValues
    }
    return settingsOrError
  }, [electionId])

  const updateSettings = async (newSettings: TNewSettings) => {
    const oldSettings = await getSettings()
    const mergedSettings = {
      ...oldSettings,
      ...newSettings,
    }
    const response: IErrorResponse = await api(
      `/election/${electionId}/settings`,
      {
        method: 'PUT',
        body: JSON.stringify(mergedSettings),
        headers: {
          'Content-Type': 'application/json',
        },
      }
    )
    if (checkAndToast(response)) {
      return
    }
    setSettings(mergedSettings)
  }

  useEffect(() => {
    ;(async () => {
      const newSettings = await getSettings()
      setSettings(newSettings)
    })()
  }, [getSettings])
  return [settings, updateSettings]
}

export default useAuditSettings
