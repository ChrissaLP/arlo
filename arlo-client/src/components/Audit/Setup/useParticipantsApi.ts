import { useState, useEffect } from 'react'
import { toast } from 'react-toastify'
import { api } from '../../utilities'

export type IJurisdictions = {
  id: string
  name: string
}[]

const useParticipantsApi = (electionId: string) => {
  const [jurisdictions, setJurisdictions] = useState<IJurisdictions>([])
  useEffect(() => {
    ;(async () => {
      try {
        const response: { jurisdictions: IJurisdictions } = await api(
          `/election/${electionId}/jurisdiction`
        )
        setJurisdictions(response.jurisdictions)
      } catch (err) {
        toast.error(err.message)
      }
    })()
  }, [electionId])
  return jurisdictions
}

export default useParticipantsApi
