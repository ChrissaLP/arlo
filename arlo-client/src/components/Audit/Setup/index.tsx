/* eslint-disable react/prop-types */
import React from 'react'
import { ElementType, IAudit } from '../../../types'
import Participants from './Participants'
import Contests from './Contests'

export const setupStages = [
  'Participants',
  'Target Contests',
  'Opportunistic Contests',
  'Audit Settings',
  'Review & Launch',
] as const

interface IProps {
  stage: ElementType<typeof setupStages>
  audit: IAudit
  setStage: (stage: ElementType<typeof setupStages>) => void
}

const Setup: React.FC<IProps> = ({ stage, setStage, audit }) => {
  const currentIndex = setupStages.indexOf(stage)
  const nextStage = () => {
    if (currentIndex < setupStages.length - 1)
      setStage(setupStages[currentIndex + 1])
  }
  const prevStage = () => {
    // modal warn that form data may be lost
    if (currentIndex > 0) setStage(setupStages[currentIndex - 1])
  }

  switch (stage) {
    case 'Participants':
      return <Participants audit={audit} nextStage={nextStage} />
    case 'Target Contests':
      return (
        <Contests
          isTargeted
          key="targeted"
          audit={audit}
          nextStage={nextStage}
          prevStage={prevStage}
        />
      )
    case 'Opportunistic Contests':
      return (
        <Contests
          isTargeted={false}
          key="opportunistic"
          audit={audit}
          nextStage={nextStage}
          prevStage={prevStage}
        />
      )
    case 'Audit Settings':
      return <p>Audit Settings</p>
    case 'Review & Launch':
      return <p>Review &amp; Launch</p>
    default:
      return null
  }
}

export default Setup
