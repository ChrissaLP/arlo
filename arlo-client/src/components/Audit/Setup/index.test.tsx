import React from 'react'
import { useParams } from 'react-router-dom'
import { statusStates, auditSettings } from '../_mocks'
import * as utilities from '../../utilities'
import { asyncActRender } from '../../testUtilities'
import Setup from './index'
import relativeStages from './_mocks'

const apiMock: jest.SpyInstance<
  ReturnType<typeof utilities.api>,
  Parameters<typeof utilities.api>
> = jest.spyOn(utilities, 'api').mockImplementation()
const checkAndToastMock: jest.SpyInstance<
  ReturnType<typeof utilities.checkAndToast>,
  Parameters<typeof utilities.checkAndToast>
> = jest.spyOn(utilities, 'checkAndToast').mockReturnValue(false)
apiMock.mockResolvedValue(auditSettings.all)

checkAndToastMock.mockReturnValue(false)

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'), // use actual for all non-hook parts
  useParams: jest.fn(),
}))
const routeMock = useParams as jest.Mock
routeMock.mockReturnValue({
  electionId: '1',
  view: 'setup',
})

afterEach(() => {
  apiMock.mockClear()
  checkAndToastMock.mockClear()
})

describe('Setup', () => {
  it('renders Participants stage', async () => {
    const { container } = await asyncActRender(
      <Setup
        audit={statusStates.sampleSizeOptions}
        stage="Participants"
        menuItems={relativeStages('Participants').menuItems}
      />
    )
    expect(container).toMatchSnapshot()
  })

  it('renders Participants stage with locked next stage', async () => {
    const { container } = await asyncActRender(
      <Setup
        audit={statusStates[2]}
        stage="Participants"
        menuItems={relativeStages('Participants', 'locked').menuItems}
      />
    )
    expect(container).toMatchSnapshot()
  })

  it('renders Participants stage with processing next stage', async () => {
    const { container } = await asyncActRender(
      <Setup
        audit={statusStates[2]}
        stage="Participants"
        menuItems={relativeStages('Participants', 'processing').menuItems}
      />
    )
    expect(container).toMatchSnapshot()
  })

  it('renders Target Contests stage', async () => {
    const { container } = await asyncActRender(
      <Setup
        audit={statusStates.sampleSizeOptions}
        stage="Target Contests"
        menuItems={relativeStages('Target Contests').menuItems}
      />
    )
    expect(container).toMatchSnapshot()
  })

  it('renders Target Contests stage with locked next stage', async () => {
    const { container } = await asyncActRender(
      <Setup
        audit={statusStates[2]}
        stage="Target Contests"
        menuItems={relativeStages('Target Contests', 'locked').menuItems}
      />
    )
    expect(container).toMatchSnapshot()
  })

  it('renders Target Contests stage with processing next stage', async () => {
    const { container } = await asyncActRender(
      <Setup
        audit={statusStates[2]}
        stage="Target Contests"
        menuItems={relativeStages('Target Contests', 'processing').menuItems}
      />
    )
    expect(container).toMatchSnapshot()
  })

  it('renders Opportunistic Contests stage', async () => {
    const { container } = await asyncActRender(
      <Setup
        audit={statusStates.sampleSizeOptions}
        stage="Opportunistic Contests"
        menuItems={relativeStages('Opportunistic Contests').menuItems}
      />
    )
    expect(container).toMatchSnapshot()
  })

  it('renders Opportunistic Contests stage with locked next stage', async () => {
    const { container } = await asyncActRender(
      <Setup
        audit={statusStates[2]}
        stage="Opportunistic Contests"
        menuItems={relativeStages('Opportunistic Contests', 'locked').menuItems}
      />
    )
    expect(container).toMatchSnapshot()
  })

  it('renders Opportunistic Contests stage with processing next stage', async () => {
    const { container } = await asyncActRender(
      <Setup
        audit={statusStates[2]}
        stage="Opportunistic Contests"
        menuItems={
          relativeStages('Opportunistic Contests', 'processing').menuItems
        }
      />
    )
    expect(container).toMatchSnapshot()
  })

  it('renders Audit Settings stage', async () => {
    const { container } = await asyncActRender(
      <Setup
        audit={statusStates.sampleSizeOptions}
        stage="Audit Settings"
        menuItems={relativeStages('Audit Settings').menuItems}
      />
    )
    expect(container).toMatchSnapshot()
  })

  it('renders Audit Settings stage with locked next stage', async () => {
    const { container } = await asyncActRender(
      <Setup
        audit={statusStates[2]}
        stage="Audit Settings"
        menuItems={relativeStages('Audit Settings', 'locked').menuItems}
      />
    )
    expect(container).toMatchSnapshot()
  })

  it('renders Audit Settings stage with processing next stage', async () => {
    const { container } = await asyncActRender(
      <Setup
        audit={statusStates[2]}
        stage="Audit Settings"
        menuItems={relativeStages('Audit Settings', 'processing').menuItems}
      />
    )
    expect(container).toMatchSnapshot()
  })

  it('renders Review & Launch stage', async () => {
    const { container } = await asyncActRender(
      <Setup
        audit={statusStates.sampleSizeOptions}
        stage="Review & Launch"
        menuItems={relativeStages('Review & Launch').menuItems}
      />
    )
    expect(container).toMatchSnapshot()
  })

  it('renders Review & Launch stage with locked next stage', async () => {
    const { container } = await asyncActRender(
      <Setup
        audit={statusStates[2]}
        stage="Review & Launch"
        menuItems={relativeStages('Review & Launch', 'locked').menuItems}
      />
    )
    expect(container).toMatchSnapshot()
  })

  it('renders Review & Launch stage with processing next stage', async () => {
    const { container } = await asyncActRender(
      <Setup
        audit={statusStates[2]}
        stage="Review & Launch"
        menuItems={relativeStages('Review & Launch', 'processing').menuItems}
      />
    )
    expect(container).toMatchSnapshot()
  })
})
