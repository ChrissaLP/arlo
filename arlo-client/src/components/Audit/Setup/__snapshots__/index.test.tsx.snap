// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Setup renders Audit Settings stage 1`] = `
<div>
  <form
    data-testid="form-one"
  >
    <div
      class="sc-bwzfXH hfheQC"
    >
      <h2
        class="bp3-heading sc-bdVaJa ibVpPq"
      >
        Audit Settings
      </h2>
      <div
        class="sc-bZQynM ibvdum"
      >
        <label
          for="election-name"
          id="election-name-label"
        >
          Election Name
          <div
            class="sc-iwsKbI dvOqsv"
          >
            <div
              class="bp3-input-group sc-gZMcBi fkRAtG"
            >
              <input
                aria-labelledby="election-name-label"
                class="bp3-input"
                id="election-name"
                name="electionName"
                style="padding-right: 10px;"
                value="Election Name"
              />
            </div>
          </div>
        </label>
      </div>
      <div
        class="sc-bZQynM ibvdum"
      >
        <label
          for="online"
        >
          Audit boards will enter data about each audited ballot:
          <div>
            <label
              class="bp3-control bp3-radio"
            >
              <input
                checked=""
                name="online"
                type="radio"
                value="online"
              />
              <span
                class="bp3-control-indicator"
              />
              Online
            </label>
            <label
              class="bp3-control bp3-radio"
            >
              <input
                name="online"
                type="radio"
                value="offline"
              />
              <span
                class="bp3-control-indicator"
              />
              Offline
            </label>
          </div>
        </label>
      </div>
      <div
        class="sc-bZQynM ibvdum"
      >
        <h3
          class="bp3-heading sc-htoDjs egbcjZ"
        >
          Desired Risk Limit
        </h3>
        <label
          for="risk-limit"
        >
          Set the risk for the audit as a percentage (e.g. "5" = 5%)
          <div
            class="bp3-html-select sc-fjdhpX jqDqiz"
          >
            <select
              data-testid="risk-limit"
              field="[object Object]"
              form="[object Object]"
              id="risk-limit"
            >
              <option>
                1
              </option>
              <option>
                2
              </option>
              <option>
                3
              </option>
              <option>
                4
              </option>
              <option>
                5
              </option>
              <option>
                6
              </option>
              <option>
                7
              </option>
              <option>
                8
              </option>
              <option>
                9
              </option>
              <option>
                10
              </option>
              <option>
                11
              </option>
              <option>
                12
              </option>
              <option>
                13
              </option>
              <option>
                14
              </option>
              <option>
                15
              </option>
              <option>
                16
              </option>
              <option>
                17
              </option>
              <option>
                18
              </option>
              <option>
                19
              </option>
              <option>
                20
              </option>
            </select>
            <span
              class="bp3-icon bp3-icon-double-caret-vertical"
              icon="double-caret-vertical"
            >
              <svg
                data-icon="double-caret-vertical"
                height="16"
                viewBox="0 0 16 16"
                width="16"
              >
                <desc>
                  double-caret-vertical
                </desc>
                <path
                  d="M5 7h6a1.003 1.003 0 00.71-1.71l-3-3C8.53 2.11 8.28 2 8 2s-.53.11-.71.29l-3 3A1.003 1.003 0 005 7zm6 2H5a1.003 1.003 0 00-.71 1.71l3 3c.18.18.43.29.71.29s.53-.11.71-.29l3-3A1.003 1.003 0 0011 9z"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
          </div>
        </label>
      </div>
      <div
        class="sc-bZQynM ibvdum"
      >
        <h3
          class="bp3-heading sc-htoDjs egbcjZ"
        >
          Random Seed
        </h3>
        <label
          for="random-seed"
          id="random-seed-label"
        >
          Enter the random characters to seed the pseudo-random number generator.
          <div
            class="sc-iwsKbI dvOqsv"
          >
            <div
              class="bp3-input-group sc-gZMcBi fkRAtG"
            >
              <input
                aria-labelledby="random-seed-label"
                class="bp3-input"
                id="random-seed"
                name="randomSeed"
                style="padding-right: 10px;"
                type="text"
                value="12345"
              />
            </div>
          </div>
        </label>
      </div>
    </div>
    <div
      class="sc-htpNat gtULUV"
    >
      <button
        class="bp3-button sc-ifAKCX zdCNs"
        type="button"
      >
        <span
          class="bp3-button-text"
        >
          Back
        </span>
      </button>
      <button
        class="bp3-button sc-ifAKCX zdCNs"
        type="button"
      >
        <span
          class="bp3-button-text"
        >
          Save & Next
        </span>
      </button>
    </div>
  </form>
</div>
`;

exports[`Setup renders Opportunistic Contests stage 1`] = `
<div>
  <form
    data-testid="form-one"
  >
    <div
      class="sc-bwzfXH hfheQC"
    >
      <h2
        class="bp3-heading sc-bdVaJa ibVpPq"
      >
        Opportunistic Contests
      </h2>
      <div
        class="sc-bZQynM ibvdum"
      >
        <h3
          class="bp3-heading sc-htoDjs egbcjZ"
        >
          Contest  Info
        </h3>
        <div
          class="sc-gzVnrw kvvweo"
        >
          Enter the name of the contest that will drive the audit.
        </div>
        <label
          for="contests[0].name"
        >
          Contest
           
          
           
          Name
          <div
            class="sc-iwsKbI dvOqsv"
          >
            <div
              class="bp3-input-group sc-gZMcBi fkRAtG"
            >
              <input
                class="bp3-input"
                id="contests[0].name"
                name="contests[0].name"
                style="padding-right: 10px;"
                value="contest name"
              />
            </div>
          </div>
        </label>
        <div
          class="sc-gzVnrw kvvweo"
        >
          Enter the number of winners for the contest.
        </div>
        <label
          for="contests[0].numWinners"
        >
          Winners
          <div
            class="sc-iwsKbI dvOqsv"
          >
            <div
              class="bp3-input-group sc-gZMcBi fkRAtG"
            >
              <input
                class="bp3-input"
                id="contests[0].numWinners"
                name="contests[0].numWinners"
                style="padding-right: 10px;"
                value="1"
              />
            </div>
          </div>
        </label>
        <div
          class="sc-gzVnrw kvvweo"
        >
          Number of selections the voter can make in the contest.
        </div>
        <label
          for="contests[0].votesAllowed"
        >
          Votes Allowed
          <div
            class="sc-iwsKbI dvOqsv"
          >
            <div
              class="bp3-input-group sc-gZMcBi fkRAtG"
            >
              <input
                class="bp3-input"
                id="contests[0].votesAllowed"
                name="contests[0].votesAllowed"
                style="padding-right: 10px;"
                value="1"
              />
            </div>
          </div>
        </label>
      </div>
      <div
        class="sc-bZQynM ibvdum"
      >
        <h3
          class="bp3-heading sc-htoDjs egbcjZ"
        >
          Candidates/Choices & Vote Totals
        </h3>
        <div
          class="sc-gzVnrw kvvweo"
        >
          Enter the name of each candidate choice that appears on the ballot for this contest.
        </div>
        <div
          class="sc-jzJRlG hqJpHa"
        >
          <div
            class="sc-kAzzGY gvpNKy"
          >
            <label
              class="bp3-label sc-kgoBCf eKfxQS"
            >
              Name of Candidate/Choice 
              1
              <div
                class="sc-chPdSV knRDOr sc-iwsKbI dvOqsv"
              >
                <div
                  class="bp3-input-group sc-gZMcBi fkRAtG"
                >
                  <input
                    class="bp3-input"
                    name="contests[0].choices[0].name"
                    style="padding-right: 10px;"
                    value="choice one"
                  />
                </div>
              </div>
            </label>
            <label
              class="bp3-label sc-kgoBCf eKfxQS"
            >
              Votes for Candidate/Choice 
              1
              <div
                class="sc-chPdSV knRDOr sc-iwsKbI dvOqsv"
              >
                <div
                  class="bp3-input-group sc-gZMcBi fkRAtG"
                >
                  <input
                    class="bp3-input"
                    name="contests[0].choices[0].numVotes"
                    style="padding-right: 10px;"
                    value="792"
                  />
                </div>
              </div>
            </label>
          </div>
          <div
            class="sc-kAzzGY gvpNKy"
          >
            <label
              class="bp3-label sc-kgoBCf eKfxQS"
            >
              Name of Candidate/Choice 
              2
              <div
                class="sc-chPdSV knRDOr sc-iwsKbI dvOqsv"
              >
                <div
                  class="bp3-input-group sc-gZMcBi fkRAtG"
                >
                  <input
                    class="bp3-input"
                    name="contests[0].choices[1].name"
                    style="padding-right: 10px;"
                    value="choice two"
                  />
                </div>
              </div>
            </label>
            <label
              class="bp3-label sc-kgoBCf eKfxQS"
            >
              Votes for Candidate/Choice 
              2
              <div
                class="sc-chPdSV knRDOr sc-iwsKbI dvOqsv"
              >
                <div
                  class="bp3-input-group sc-gZMcBi fkRAtG"
                >
                  <input
                    class="bp3-input"
                    name="contests[0].choices[1].numVotes"
                    style="padding-right: 10px;"
                    value="1325"
                  />
                </div>
              </div>
            </label>
          </div>
        </div>
      </div>
      <div
        class="sc-bZQynM ibvdum"
      >
        <h3
          class="bp3-heading sc-htoDjs egbcjZ"
        >
          Total Ballots Cast
        </h3>
        <div
          class="sc-gzVnrw kvvweo"
        >
          Enter the overall number of ballot cards cast in jurisdictions containing this contest.
        </div>
        <label
          for="contests[0].totalBallotsCast"
        >
          Total Ballots for Contest
           
          
          <div
            class="sc-iwsKbI dvOqsv"
          >
            <div
              class="bp3-input-group sc-gZMcBi fkRAtG"
            >
              <input
                class="bp3-input"
                id="contests[0].totalBallotsCast"
                name="contests[0].totalBallotsCast"
                style="padding-right: 10px;"
                value="2123"
              />
            </div>
          </div>
        </label>
      </div>
    </div>
  </form>
</div>
`;

exports[`Setup renders Participants stage 1`] = `
<div>
  <form
    data-testid="form-one"
  >
    <div
      class="sc-bwzfXH hfheQC"
    >
      <h2
        class="bp3-heading sc-bdVaJa ibVpPq"
      >
        Participants
      </h2>
      <label
        for="state"
      >
        Choose your state from the options below
        <br />
        <div
          class="bp3-html-select sc-dnqmqq kLRTmU"
        >
          <select
            data-testid="state-field"
            field="[object Object]"
            form="[object Object]"
            id="state"
          >
            <option
              value=""
            />
            <option
              label="Alabama"
              value="AL"
            >
              Alabama
            </option>
            <option
              label="Alaska"
              value="AK"
            >
              Alaska
            </option>
            <option
              label="Arizona"
              value="AZ"
            >
              Arizona
            </option>
            <option
              label="Arkansas"
              value="AR"
            >
              Arkansas
            </option>
            <option
              label="California"
              value="CA"
            >
              California
            </option>
            <option
              label="Colorado"
              value="CO"
            >
              Colorado
            </option>
            <option
              label="Connecticut"
              value="CT"
            >
              Connecticut
            </option>
            <option
              label="Delaware"
              value="DE"
            >
              Delaware
            </option>
            <option
              label="District Of Columbia"
              value="DC"
            >
              District Of Columbia
            </option>
            <option
              label="Florida"
              value="FL"
            >
              Florida
            </option>
            <option
              label="Georgia"
              value="GA"
            >
              Georgia
            </option>
            <option
              label="Hawaii"
              value="HI"
            >
              Hawaii
            </option>
            <option
              label="Idaho"
              value="ID"
            >
              Idaho
            </option>
            <option
              label="Illinois"
              value="IL"
            >
              Illinois
            </option>
            <option
              label="Indiana"
              value="IN"
            >
              Indiana
            </option>
            <option
              label="Iowa"
              value="IA"
            >
              Iowa
            </option>
            <option
              label="Kansas"
              value="KS"
            >
              Kansas
            </option>
            <option
              label="Kentucky"
              value="KY"
            >
              Kentucky
            </option>
            <option
              label="Louisiana"
              value="LA"
            >
              Louisiana
            </option>
            <option
              label="Maine"
              value="ME"
            >
              Maine
            </option>
            <option
              label="Maryland"
              value="MD"
            >
              Maryland
            </option>
            <option
              label="Massachusetts"
              value="MA"
            >
              Massachusetts
            </option>
            <option
              label="Michigan"
              value="MI"
            >
              Michigan
            </option>
            <option
              label="Minnesota"
              value="MN"
            >
              Minnesota
            </option>
            <option
              label="Mississippi"
              value="MS"
            >
              Mississippi
            </option>
            <option
              label="Missouri"
              value="MO"
            >
              Missouri
            </option>
            <option
              label="Montana"
              value="MT"
            >
              Montana
            </option>
            <option
              label="Nebraska"
              value="NE"
            >
              Nebraska
            </option>
            <option
              label="Nevada"
              value="NV"
            >
              Nevada
            </option>
            <option
              label="New Hampshire"
              value="NH"
            >
              New Hampshire
            </option>
            <option
              label="New Jersey"
              value="NJ"
            >
              New Jersey
            </option>
            <option
              label="New Mexico"
              value="NM"
            >
              New Mexico
            </option>
            <option
              label="New York"
              value="NY"
            >
              New York
            </option>
            <option
              label="North Carolina"
              value="NC"
            >
              North Carolina
            </option>
            <option
              label="North Dakota"
              value="ND"
            >
              North Dakota
            </option>
            <option
              label="Ohio"
              value="OH"
            >
              Ohio
            </option>
            <option
              label="Oklahoma"
              value="OK"
            >
              Oklahoma
            </option>
            <option
              label="Oregon"
              value="OR"
            >
              Oregon
            </option>
            <option
              label="Pennsylvania"
              value="PA"
            >
              Pennsylvania
            </option>
            <option
              label="Rhode Island"
              value="RI"
            >
              Rhode Island
            </option>
            <option
              label="South Carolina"
              value="SC"
            >
              South Carolina
            </option>
            <option
              label="South Dakota"
              value="SD"
            >
              South Dakota
            </option>
            <option
              label="Tennessee"
              value="TN"
            >
              Tennessee
            </option>
            <option
              label="Texas"
              value="TX"
            >
              Texas
            </option>
            <option
              label="Utah"
              value="UT"
            >
              Utah
            </option>
            <option
              label="Vermont"
              value="VT"
            >
              Vermont
            </option>
            <option
              label="Virginia"
              value="VA"
            >
              Virginia
            </option>
            <option
              label="Washington"
              value="WA"
            >
              Washington
            </option>
            <option
              label="West Virginia"
              value="WV"
            >
              West Virginia
            </option>
            <option
              label="Wisconsin"
              value="WI"
            >
              Wisconsin
            </option>
            <option
              label="Wyoming"
              value="WY"
            >
              Wyoming
            </option>
          </select>
          <span
            class="bp3-icon bp3-icon-double-caret-vertical"
            icon="double-caret-vertical"
          >
            <svg
              data-icon="double-caret-vertical"
              height="16"
              viewBox="0 0 16 16"
              width="16"
            >
              <desc>
                double-caret-vertical
              </desc>
              <path
                d="M5 7h6a1.003 1.003 0 00.71-1.71l-3-3C8.53 2.11 8.28 2 8 2s-.53.11-.71.29l-3 3A1.003 1.003 0 005 7zm6 2H5a1.003 1.003 0 00-.71 1.71l3 3c.18.18.43.29.71.29s.53-.11.71-.29l3-3A1.003 1.003 0 0011 9z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
        </div>
      </label>
      <div
        class="sc-bZQynM ibvdum"
      >
        <div
          class="sc-gzVnrw kvvweo"
        >
          Click "Browse" to choose the appropriate file from your computer. This file should be a comma-separated list of all the jurisdictions participating in the audit, plus email addresses for audit administrators in each participating jurisdiction.
          <br />
          <br />
          <a
            href="http://localhost/sample_jurisdiction_filesheet.csv"
            rel="noopener noreferrer"
            target="_blank"
          >
            (Click here to view a sample file in the correct format.)
          </a>
        </div>
      </div>
      <div
        class="sc-bZQynM ibvdum"
      >
        <label
          class="bp3-file-input"
        >
          <input
            accept=".csv"
            name="csv"
            type="file"
          />
          <span
            class="bp3-file-upload-input"
          >
            Select a CSV...
          </span>
        </label>
      </div>
    </div>
    <div
      class="sc-htpNat gtULUV"
    >
      <button
        class="bp3-button bp3-intent-primary sc-ifAKCX zdCNs"
        type="submit"
      >
        <span
          class="bp3-button-text"
        >
          Save & Next
        </span>
      </button>
    </div>
  </form>
</div>
`;

exports[`Setup renders Review & Launch stage 1`] = `
<div>
  <div>
    <p>
      Review
    </p>
    <div
      class="sc-htpNat gtULUV"
    >
      <button
        class="bp3-button sc-ifAKCX zdCNs"
        type="button"
      >
        <span
          class="bp3-button-text"
        >
          Back
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`Setup renders Target Contests stage 1`] = `
<div>
  <form
    data-testid="form-one"
  >
    <div
      class="sc-bwzfXH hfheQC"
    >
      <h2
        class="bp3-heading sc-bdVaJa ibVpPq"
      >
        Target Contests
      </h2>
      <div
        class="sc-bZQynM ibvdum"
      >
        <h3
          class="bp3-heading sc-htoDjs egbcjZ"
        >
          Contest  Info
        </h3>
        <div
          class="sc-gzVnrw kvvweo"
        >
          Enter the name of the contest that will drive the audit.
        </div>
        <label
          for="contests[0].name"
        >
          Contest
           
          
           
          Name
          <div
            class="sc-iwsKbI dvOqsv"
          >
            <div
              class="bp3-input-group sc-gZMcBi fkRAtG"
            >
              <input
                class="bp3-input"
                id="contests[0].name"
                name="contests[0].name"
                style="padding-right: 10px;"
                value="contest name"
              />
            </div>
          </div>
        </label>
        <div
          class="sc-gzVnrw kvvweo"
        >
          Enter the number of winners for the contest.
        </div>
        <label
          for="contests[0].numWinners"
        >
          Winners
          <div
            class="sc-iwsKbI dvOqsv"
          >
            <div
              class="bp3-input-group sc-gZMcBi fkRAtG"
            >
              <input
                class="bp3-input"
                id="contests[0].numWinners"
                name="contests[0].numWinners"
                style="padding-right: 10px;"
                value="1"
              />
            </div>
          </div>
        </label>
        <div
          class="sc-gzVnrw kvvweo"
        >
          Number of selections the voter can make in the contest.
        </div>
        <label
          for="contests[0].votesAllowed"
        >
          Votes Allowed
          <div
            class="sc-iwsKbI dvOqsv"
          >
            <div
              class="bp3-input-group sc-gZMcBi fkRAtG"
            >
              <input
                class="bp3-input"
                id="contests[0].votesAllowed"
                name="contests[0].votesAllowed"
                style="padding-right: 10px;"
                value="1"
              />
            </div>
          </div>
        </label>
      </div>
      <div
        class="sc-bZQynM ibvdum"
      >
        <h3
          class="bp3-heading sc-htoDjs egbcjZ"
        >
          Candidates/Choices & Vote Totals
        </h3>
        <div
          class="sc-gzVnrw kvvweo"
        >
          Enter the name of each candidate choice that appears on the ballot for this contest.
        </div>
        <div
          class="sc-jzJRlG hqJpHa"
        >
          <div
            class="sc-kAzzGY gvpNKy"
          >
            <label
              class="bp3-label sc-kgoBCf eKfxQS"
            >
              Name of Candidate/Choice 
              1
              <div
                class="sc-chPdSV knRDOr sc-iwsKbI dvOqsv"
              >
                <div
                  class="bp3-input-group sc-gZMcBi fkRAtG"
                >
                  <input
                    class="bp3-input"
                    name="contests[0].choices[0].name"
                    style="padding-right: 10px;"
                    value="choice one"
                  />
                </div>
              </div>
            </label>
            <label
              class="bp3-label sc-kgoBCf eKfxQS"
            >
              Votes for Candidate/Choice 
              1
              <div
                class="sc-chPdSV knRDOr sc-iwsKbI dvOqsv"
              >
                <div
                  class="bp3-input-group sc-gZMcBi fkRAtG"
                >
                  <input
                    class="bp3-input"
                    name="contests[0].choices[0].numVotes"
                    style="padding-right: 10px;"
                    value="792"
                  />
                </div>
              </div>
            </label>
          </div>
          <div
            class="sc-kAzzGY gvpNKy"
          >
            <label
              class="bp3-label sc-kgoBCf eKfxQS"
            >
              Name of Candidate/Choice 
              2
              <div
                class="sc-chPdSV knRDOr sc-iwsKbI dvOqsv"
              >
                <div
                  class="bp3-input-group sc-gZMcBi fkRAtG"
                >
                  <input
                    class="bp3-input"
                    name="contests[0].choices[1].name"
                    style="padding-right: 10px;"
                    value="choice two"
                  />
                </div>
              </div>
            </label>
            <label
              class="bp3-label sc-kgoBCf eKfxQS"
            >
              Votes for Candidate/Choice 
              2
              <div
                class="sc-chPdSV knRDOr sc-iwsKbI dvOqsv"
              >
                <div
                  class="bp3-input-group sc-gZMcBi fkRAtG"
                >
                  <input
                    class="bp3-input"
                    name="contests[0].choices[1].numVotes"
                    style="padding-right: 10px;"
                    value="1325"
                  />
                </div>
              </div>
            </label>
          </div>
        </div>
      </div>
      <div
        class="sc-bZQynM ibvdum"
      >
        <h3
          class="bp3-heading sc-htoDjs egbcjZ"
        >
          Total Ballots Cast
        </h3>
        <div
          class="sc-gzVnrw kvvweo"
        >
          Enter the overall number of ballot cards cast in jurisdictions containing this contest.
        </div>
        <label
          for="contests[0].totalBallotsCast"
        >
          Total Ballots for Contest
           
          
          <div
            class="sc-iwsKbI dvOqsv"
          >
            <div
              class="bp3-input-group sc-gZMcBi fkRAtG"
            >
              <input
                class="bp3-input"
                id="contests[0].totalBallotsCast"
                name="contests[0].totalBallotsCast"
                style="padding-right: 10px;"
                value="2123"
              />
            </div>
          </div>
        </label>
      </div>
    </div>
  </form>
</div>
`;
